zookeeper-metrics-providers/zookeeper-prometheus-metrics/src/test/java/org/apache/zookeeper/metrics/prometheus/PrometheusMetricsProviderConfigTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.metrics.prometheus;
+
+import io.prometheus.client.CollectorRegistry;
+
+import java.util.Properties;
+
+import org.apache.zookeeper.metrics.MetricsProviderLifeCycleException;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class PrometheusMetricsProviderConfigTest {
+
+    @Test
+    public void testInvalidPort() {
+        MetricsProviderLifeCycleException exception = Assert
+                .assertThrows(MetricsProviderLifeCycleException.class, () -> {
+                    CollectorRegistry.defaultRegistry.clear();
+                    PrometheusMetricsProvider provider = new PrometheusMetricsProvider();
+                    Properties configuration = new Properties();
+                    configuration.setProperty("httpPort", "65536");
+                    configuration.setProperty("exportJvmInfo", "false");
+                    provider.configure(configuration);
+                    provider.start();
+                });
+        Assert.assertEquals("java.lang.IllegalArgumentException: port out of range:65536", exception.getMessage());
+    }
+
+    @Test
+    public void testInvalidAddr() {
+        MetricsProviderLifeCycleException exception = Assert
+                .assertThrows(MetricsProviderLifeCycleException.class, () -> {
+                    CollectorRegistry.defaultRegistry.clear();
+                    PrometheusMetricsProvider provider = new PrometheusMetricsProvider();
+                    Properties configuration = new Properties();
+                    configuration.setProperty("httpHost", "256.256.256.256");
+                    provider.configure(configuration);
+                    provider.start();
+                });
+        Assert.assertEquals("java.net.SocketException: Unresolved address", exception.getMessage());
```

Comment:
- You have asserted there was a `MetricsProviderLifeCycleException` happens here, so don't make so specify assertion here to avoid the environment diversity？So removing this line?

Reply:
Removed specify assertion